lib/core/src/Cardano/Wallet/DB.hs

```diff
@@ -301,12 +302,13 @@ data DBLayer m s k = forall stm. (MonadIO stm, MonadFail stm) => DBLayer
 
     , rollbackTo
         :: WalletId
-        -> SlotNo
-        -> ExceptT ErrNoSuchWallet stm BlockHeader
-        -- ^ Drops all checkpoints and transaction data after the given slot.
+        -> Slot
```

Comment:
I agree that this will not happen in practice on testnet and mainnet. However, the ouroboros-network library does represent points on the blockchain with an explicit origin, and a rollback to genesis can happen in principle (and does happen on local test clusters). By using a representation that is more closely aligned with ouroboros-network (i.e. `Slot` instead of `SlotNo`), the code actually becomes simpler, as we can remove conversion functions such as

```haskell
    pseudoPointSlot :: ChainPoint -> SlotNo
    pseudoPointSlot ChainPointAtGenesis = W.SlotNo 0
    pseudoPointSlot (ChainPoint slot _) = slot
```